### DISCUSSION TITLE
Plotting needs 24+ hours

### DISCUSSION THREAD

**Discussion by Stinner1009** [Chia-Network/chia-blockchain#6233] [Support]
Can anyone help me figure out why plotting needs such a long time ?

System: 2x AMD EPYC 7742 64-Core
             1 TB RAM 3200MHZ
             70TB Samsung MZILT7T6HALA/007 SCSI

It doesn`t depends on the amount of parallel plots.
Maybe I got to boost the CPU Clock speed?

any ideas?

**Comment by XIII39** [comment]
What is your clock speed currently?
What is your plot speed if you run a single plot, 4 threads, 3500 mem? Is there a phase that is particularly slow?

Keep an eye on your SSD temps, I've been finding that if the drive temps exceed 60-65 degrees Celsius, plotting slows way down. My solution so far has been to slap a thermal pad onto the drive to help dissipate the heat.

**Reply by Stinner1009** [reply]
Clock speed is most time between 1,8GHz and 2,1 GHz.
tried to plot with 2 ,4 and 6 threads. no big impact on plotting time.
there`s no phase that is particuarly slow. it`s more like everey phase needs 4-5 times longer as it should take.

temperatures are definitly not the problem.

**Reply by XIII39** [reply]
I can't speak directly to your CPU overclock settings, I'm sure someone on here is much more versed on the subject. Me and AMD processors are not the best of friends..

I generally see about 5 hour plot times, running 8 in parallel with a 45 minute stagger (two 2tb SSD temp drives) sustaining 4.6 MHz on each core of my i7 10700k and running my memory at 3600 MHz (two Kingston 16gb sticks, XMP I settings in the BIOS)

**Reply by Ufkabakan** [reply]
> 
> 
> I can't speak directly to your CPU overclock settings, I'm sure someone on here is much more versed on the subject. Me and AMD processors are not the best of friends..
> 
> I generally see about 5 hour plot times, running 8 in parallel with a 45 minute stagger (two 2tb SSD temp drives) sustaining 4.6 MHz on each core of my i7 10700k and running my memory at 3600 MHz (two Kingston 16gb sticks, XMP I settings in the BIOS)

How much plot can you create in a day with a 10700K Harvester PC? 24/5 * 8 = 38 ~ plot per day?
I have very similar setup, i can create 28~ in a day. (10700K 5Ghz AllCore, 32GB 3200Mhz Ram, 2x2TB 970 Evo Plus)

**Reply by XIII39** [reply]
I'm averaging about 31 plots per day. Honestly I haven't been able to run my plotters for more than 48 strait hours as something always comes up and I need to re-config something or restart a system for one reason or another. I am anxious for a week where I can run continually without interruption and avoid the 5-6 hour ramp-up time. I think once everything is running smoothly, I should be able to achieve 38, maybe even 40 plots per day but currently.. the ambient temps in my little computer room in my house are posing to be an issue. I have quite a bit of hardware piling up in the room and it's all starting to make a little too much heat which in turn is slowing things down as well.

**Reply by starpars** [reply]
> 
> 
> I'm averaging about 31 plots per day. Honestly I haven't been able to run my plotters for more than 48 strait hours as something always comes up and I need to re-config something or restart a system for one reason or another. I am anxious for a week where I can run continually without interruption and avoid the 5-6 hour ramp-up time. I think once everything is running smoothly, I should be able to achieve 38, maybe even 40 plots per day but currently.. the ambient temps in my little computer room in my house are posing to be an issue. I have quite a bit of hardware piling up in the room and it's all starting to make a little too much heat which in turn is slowing things down as well.

What is the model of your ssd nvme?

**Reply by XIII39** [reply]
I'm using WD Black SN750 on one plotter and Samsung 970 EVO on another. Samsung is faster but the WD keeps up
I also add a strip of 1.5mm thermal pad to the top of each SSD to try to keep the temps at bay. I use the ASSU TUF Z590-Plus mobo, but the thermal pads on the stock 'heatsink', if you can call it a heatsink, are garbage. I've had good results with the Extreme Odyssey pads from Thermalright.

**Reply by starpars** [reply]
I also use the same motherboard، I also use the fan to cool the ssd so that the temperature did not rise above 46 degrees.
In general, stock heatsinks are not only efficient in this work. If the fan is used, we will not have any worries about the temperature.

**Reply by kindofblue** [reply]
Best way is to give the hardware to me and let me plot for you. lol

**Reply by room101-dev** [reply]
> Clock speed is most time between 1,8GHz and 2,1 GHz.
> tried to plot with 2 ,4 and 6 threads. no big impact on plotting time.
> there`s no phase that is particuarly slow. it`s more like everey phase needs 4-5 times longer as it should take.
> 
> temperatures are definitly not the problem.

U still never told us what you were doing. Let's say your real, you have 1TB dynram@2666, then you could create a ram-disk of say 600gb, and to two plots 2x parallel, but no more; also I have found from personal testing the plotting in RAM is slower, than plotting on a NVME M.2 drive, reason is the cpu controller, doesn't have time to fetch, so the bottle-neck ends up being the memory-controller, in my tests the ram-disk was actually slower.
Lastly, you could also allocate 900GB and do 3x, but I think your system would grind to a halt, each plots is 1.5Trillion random r/w blocks the cpu would handle one plot, maybe two, three you would never perform.

I think if you study this crap, you can see that everybody plots on NMVE, temp storage on sata-ssd, and final on 7200rpm hdd; The reason is this works and throughput is 6hrs per plot.

**Comment by room101-dev** [comment]
> Can anyone help me figure out why plotting needs such a long time ?
> 
> System: 2x AMD EPYC 7742 64-Core
> 1 TB RAM 3200MHZ
> 70TB Samsung MZILT7T6HALA/007 SCSI
> 
> It doesn`t depends on the amount of parallel plots.
> Maybe I got to boost the CPU Clock speed?
> 
> any ideas?

well u have to stagger each 'plot' by > 30min, if you say ran all at once, then they would all try to write the final plot file, also I note here your not running

'chia create -t nvem -2 sata -d hdd'

U must have the hdd sata -2 as intermediate

if yOU run all at once, it could take days to write the files , as opposed to one at a time about 20min on average for 108gb at 100mb/sec
the entire purpose of -2 is to get the final off the nvme fast, then start the next plot, and let the another process write to hdd

**U need ... a cpu with lots of memory
lots of 1-2TB NVME evo980 class drives for -t
a few 1-2tb sata evo 980's for -2**

and a zillion 10+tb hdd's 7700rmp for plot storage, all physcially connected to 'farmers', no lan, or usb allowed to farm with new 5 second rule

**Comment by dysuwen** [comment]
AMD 3960X Single Thread Rating: 2686
i use Intel P3700 2TB *3,  21 Parallel task (K32 2 threads) time is 8-9 hours

AMD 7742 Single Thread Rating: 2220
2*7742 The maximum number of tasks does not exceed 128, the theoretical time should be 9-10 hours

1. It is recommended that SSD use pass-through mode and enable disk caching, but do not enable RAID card write-back
2. The task setting is best for 1 SSD to correspond to 1 or more HDDs
3. If you using linux , please observe the wa (iowait) value of the top command. Long-term greater than 20% means that the SSD has a bottleneck.
4. Check the motherboard BIOS power management mode to ensure that it is in performance mode

My English is not good, part of it is machine translation。

**Comment by BrandtH22** [comment]
@Stinner1009 , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
